From 02195e9f352ffec7d2fef851af6d74875da1c93b Mon Sep 17 00:00:00 2001
From: Vishal Sagar <vishal.sagar@xilinx.com>
Date: Thu, 3 Sep 2020 12:05:48 -0700
Subject: [PATCH 1559/1852] v4l: xilinx: sdirxss: Update the xfer_func

commit 34f095d07d2f473da27045b08854c48d646bffc5 from
https://github.com/Xilinx/linux-xlnx.git

Fix the transfer function in the v4l2_mbus_framefmt to correct values
based on the colorimetry and EOTF decoded from the ST352 payload.

Signed-off-by: Vishal Sagar <vishal.sagar@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 drivers/media/platform/xilinx/xilinx-sdirxss.c | 4 ++++
 1 file changed, 4 insertions(+)

diff --git a/drivers/media/platform/xilinx/xilinx-sdirxss.c b/drivers/media/platform/xilinx/xilinx-sdirxss.c
index a47fc2d15922..5fbfb8729d11 100644
--- a/drivers/media/platform/xilinx/xilinx-sdirxss.c
+++ b/drivers/media/platform/xilinx/xilinx-sdirxss.c
@@ -1355,6 +1355,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 
 	state->static_hdr.eotf = V4L2_EOTF_TRADITIONAL_GAMMA_SDR;
 	format->colorspace = V4L2_COLORSPACE_SMPTE170M;
+	format->xfer_func = V4L2_XFER_FUNC_709;
 
 	if (mode != XSDIRX_MODE_SD_MASK) {
 		u8 eotf = (payload & XST352_BYTE2_EOTF_MASK) >>
@@ -1371,9 +1372,11 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 			break;
 		case XST352_BYTE2_EOTF_SMPTE2084:
 			state->static_hdr.eotf = V4L2_EOTF_SMPTE_ST2084;
+			format->xfer_func = V4L2_XFER_FUNC_SMPTE2084;
 			break;
 		case XST352_BYTE2_EOTF_HLG:
 			state->static_hdr.eotf = V4L2_EOTF_BT_2100_HLG;
+			format->xfer_func = V4L2_XFER_FUNC_HLG;
 			break;
 		}
 
@@ -1391,6 +1394,7 @@ static int xsdirx_get_stream_properties(struct xsdirxss_state *state)
 			 * are currently not supported
 			 */
 			format->colorspace = V4L2_COLORSPACE_DEFAULT;
+			format->xfer_func = V4L2_XFER_FUNC_DEFAULT;
 			break;
 		}
 	}
-- 
2.31.1

